semanticdb/semanticdb3/guide.md

```diff
@@ -0,0 +1,576 @@
+_empty_.Test.main(Array).(args) => param args: Array[String]
+  Array => scala.Array#
+  String => scala.Predef.String#
+_root_. => package _root_
+```
+
+Note how the resulting SemanticDB payload is missing the `Text` attribute and
+the `Occurrences` section. That's because Metacp doesn't know anything about
+sources and only works with classes.
+
+<table>
+  <tr>
+    <td width="140px">Option</td>
+    <td>Value</td>
+    <td>Explanation</td>
+    <td>Default</td>
+  </tr>
+  <tr>
+    <td></td>
+    <td>Java classpath</td>
+    <td>
+      Specifies classpath to be converted to SemanticDB. Can contain classfiles
+      created both by the Scala compiler and the Java compiler. In both cases,
+      Metacp produces an adequate representation of the metadata.
+    </td>
+    <td></td>
+  </tr>
+  <tr>
+    <td><code>--cache-dir</code></td>
+    <td>Absolute or relative path</td>
+    <td>
+      Says where Metacp should cache results of converting jar files.
+      In order to speed up conversion, Metacp fingerprints jars and then
+      caches their converted counterparts in a configurable directory.
+      This only works for jars - class directories are never cached.
+    </td>
+    <td>
+      Platform-specific user-specific cache directory whose name includes
+      <code>semanticdb</code> and version number of Metacp.
+    </td>
+  </tr>
+  <tr>
+    <td>
+      <code>--exclude-scala-library-synthetics</code>,<br/>
+      <code>--include-scala-library-synthetics</code>
+    </td>
+    <td></td>
+    <td>
+      Specifies whether the output classpath should include a jar that contains
+      SemanticDB files corresponding to definitions missing from
+      <code>scala-library.jar</code>, e.g. <code>scala.Any</code>,
+      <code>scala.AnyRef</code> and others.
+    </td>
+    <td>
+      <code>--exclude-scala-library-synthetics</code>
+    </td>
+  </tr>
+  <tr>
+    <td><code>--par</code>,<br/><code>--no-par</code></td>
+    <td></td>
+    <td>
+      Toggles parallel processing. If enabled, classpath entries
+      will be converted in parallel according to the strategy chosen
+      by parallel collections of the Scala standard library.
+      <br/><br/>
+      NOTE: Some of our users <a href="https://github.com/scalameta/scalameta/issues/1398">have reported deadlocks</a> supposedly caused
+      by enabling <code>--par</code>. Proceed at your own risk.
+    </td>
+    <td>
+      <code>--no-par</code>
+    </td>
+  </tr>
+</table>
+
+## Consuming SemanticDB
+
+### Scalameta
+
+Scalameta includes the `semanticdb3` library that contains
+[ScalaPB](https://scalapb.github.io/) bindings to
+[the SemanticDB protobuf schema](semanticdb3.proto). Using this library,
+one can model SemanticDB entities as Scala case classes and serialize/deserialize
+them into bytes and streams. For more information, check out [the Scaladoc](https://static.javadoc.io/org.scalameta/semanticdb3_2.11/3.7.4/index.html#scala.meta.internal.semanticdb3.TextDocuments).
```

Comment:
Link to 2.12 docs instead? https://static.javadoc.io/org.scalameta/semanticdb3_2.12/3.7.4/scala/meta/internal/semanticdb3/TextDocuments.html